indexer-balances/src/cache.rs

```diff
@@ -0,0 +1,52 @@
+use cached::{Cached, SizedCache};
+
+use near_lake_framework::near_indexer_primitives;
+use tokio::sync::Mutex;
+
+use crate::metrics;
+
+pub struct BalanceCache {
+    cache: std::sync::Arc<
+        Mutex<SizedCache<near_indexer_primitives::types::AccountId, crate::BalanceDetails>>,
+    >,
+}
+
+impl BalanceCache {
```

Comment:
Extracted this out for two reasons:
1. This enforces better management of locks, i.e. lock is dropped immediately after value is returned. Previously we would wait for the balance (RPC) request to complete before dropping the lock.
2. We can encapsulate the new cache metrics which are being published.